File: hojun/FullScan/IslandPerimeter.ts
function islandPerimeter(grid: number[][]): number {
    let answer = 0;
    const axisXLength = grid[0].length;
    const voidArray = Array(axisXLength + 2).fill(0);

    const newGrid = [voidArray, ...grid.map((array) => [0, ...array, 0]), voidArray];

    newGrid.forEach((array, row) => {
        array.forEach((element, index) => {
            if (element === 1) {
                answer +=
                    4 -
                    (newGrid[row - 1][index] +
                        newGrid[row + 1][index] +
                        newGrid[row][index - 1] +
                        newGrid[row][index + 1]);
            }
        });
    });

    return answer;
}

File: dhChoi/src/Q22_06_05/IslandPerimeter.java
package Q22_06_05;

public class IslandPerimeter {

  public int islandPerimeter(int[][] grid) {
    int count = 0;
    for (int i = 0; i < grid.length; i++) {
      for (int j = 0; j < grid[i].length; j++) {
        if (grid[i][j] == 1) {
          count += 4;
          if (j > 0 && grid[i][j - 1] == 1) {
            count--;
          }
          if (j < grid[i].length - 1 && grid[i][j + 1] == 1) {
            count--;
          }
          if (i > 0 && grid[i - 1][j] == 1) {
            count--;
          }
          if (i < grid.length - 1 && grid[i + 1][j] == 1) {
            count--;
          }
        }
      }
    }
    return count;
  }
}
